src/parser/name_resolution.c

```diff
@@ -243,7 +251,139 @@ static PT_NODE *pt_get_attr_list_of_derived_table (PARSER_CONTEXT * parser, PT_M
 static void pt_set_attr_list_types (PARSER_CONTEXT * parser, PT_NODE * as_attr_list, PT_MISC_TYPE derived_table_type,
 				    PT_NODE * derived_table, PT_NODE * parent_spec);
 static PT_NODE *pt_count_with_clauses (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk);
-static int pt_resolve_dblink_server_name (PARSER_CONTEXT * parser, PT_NODE * node);
+
+static int pt_resolve_dblink_server_name (PARSER_CONTEXT * parser, PT_NODE * node, char **server_owner_name);
+static int pt_resolve_dblink_check_owner_name (PARSER_CONTEXT * parser, PT_NODE * node, char **server_owner_name);
+
+static void pt_gather_dblink_colums (PARSER_CONTEXT * parser, PT_NODE * query_stmt);
+typedef struct
+{
+  int norder;
+  int name_pos;
+  int type_idx;
+  int dec_precision;
+  int precision;
+} S_REMOTE_COL_ATTR;
+
+typedef struct remote_tbl_cols S_REMOTE_TBL_COLS;
+struct remote_tbl_cols
+{
+private:
+  int m_nm_alloc;
+  int m_nm_used;
+  char *m_nm_buf;
+
+  int m_attr_alloc;
+  int m_attr_used;
+  S_REMOTE_COL_ATTR *m_attr;
+
+private:
+    bool alloc_name_buffer (int len)
+  {
+    if (m_nm_alloc > (m_nm_used + len))
+      {
+	return true;
+      }
+
+    while (m_nm_alloc <= (m_nm_used + len))
+      {
+	m_nm_alloc += 1024;
+      }
+
+    m_nm_buf = (char *) (m_nm_buf ? realloc (m_nm_buf, m_nm_alloc) : malloc (m_nm_alloc));
+    return (m_nm_buf != NULL);
+  }
+
+  bool alloc_attr_buffer ()
+  {
+    int base_size = 10;
+    int incr_size = 2;
+
+    if (m_attr == NULL || m_attr_alloc == 0)
+      {
+	m_attr_used = m_nm_used = 0;
+	m_attr_alloc = base_size;
+	m_attr = (S_REMOTE_COL_ATTR *) malloc (m_attr_alloc * sizeof (S_REMOTE_COL_ATTR));
+      }
+    else
+      {
+	if (m_attr_alloc <= m_attr_used)
+	  {
+	    m_attr = (S_REMOTE_COL_ATTR *) realloc (m_attr, sizeof (S_REMOTE_COL_ATTR) * (m_attr_used + incr_size));
+	    m_attr_alloc = m_attr_used + incr_size;
+	  }
+      }
+
+    return (m_attr != NULL);
+  }
+
+public:
+  remote_tbl_cols ()
+  {
+    m_nm_alloc = m_nm_used = 0;
+    m_nm_buf = NULL;
+    m_attr_alloc = m_attr_used = 0;
+    m_attr = NULL;
+  }
+
+  ~remote_tbl_cols ()
+  {
+    if (m_nm_buf)
+      {
+	free (m_nm_buf);
+      }
+    if (m_attr)
+      {
+	free (m_attr);
+      }
+  }
+
```

Comment:
Make the struct non-copyable.
```suggestion
  remote_tbl_cols (const remote_tbl_cols &) = delete;
  remote_tbl_cols& operator=(const remote_tbl_cols &) = delete;
```

Reply:
Ok.